src/services/services.ts

```diff
@@ -3273,12 +3273,15 @@ export function getPropertySymbolsFromContextualType(node: ObjectLiteralElementW
         return symbol ? [symbol] : emptyArray;
     }
 
-    const discriminatedPropertySymbols = mapDefined(contextualType.types, t => (isObjectLiteralExpression(node.parent) || isJsxAttributes(node.parent)) && checker.isTypeInvalidDueToUnionDiscriminant(t, node.parent) ? undefined : t.getProperty(name));
+    const filteredTypes = isObjectLiteralExpression(node.parent) || isJsxAttributes(node.parent)
+        ? filter(contextualType.types, t => !checker.isTypeInvalidDueToUnionDiscriminant(t, node.parent))
+        : contextualType.types;
+    const discriminatedPropertySymbols = mapDefined(filteredTypes, t => t.getProperty(name));
     if (unionSymbolOk && (discriminatedPropertySymbols.length === 0 || discriminatedPropertySymbols.length === contextualType.types.length)) {
         const symbol = contextualType.getProperty(name);
         if (symbol) return [symbol];
     }
-    if (discriminatedPropertySymbols.length === 0) {
+    if (!filteredTypes.length && !discriminatedPropertySymbols.length) {

```

Comment:
The main idea behind the fix is that we don't want to go into this branch because the type satisfies its contextual type in this test case. So there is one filtered type here **but** there are no discriminated property symbols. The first type in the union doesn't have an explicit prop for `someProp` and the second one is rejected by the `isTypeInvalidDueToUnionDiscriminant` check.